CHANGELOG.md

```diff
@@ -1,5 +1,51 @@
 # Changelog
 
+## 16.4.0 (09/18/2024)
+
+### Machine ID for Terraform Cloud
```

Comment:
@timothyb89 should we say `Terraform Enterprise` instead?

Can we add the docs link here too?

Reply:
Ah, hmm. I think officially the name is HCP Terraform, and Terraform Enterprise is the separate self hosted version of that, which we also support. They renamed it from Terraform Cloud earlier this year.

The v16 docs backport is in the merge queue. They're available at https://goteleport.com/docs/ver/17.x/admin-guides/infrastructure-as-code/terraform-provider/terraform-cloud/ for the moment.

Reply:
That is a really awkward limitation that we should fix. The changelog is one of the most important places to have docs links!

cc @xinding33 @ptgott can we make this a docs team goal for Q4?

Reply:
Yep, agreed.